src/event/event_queue.rs

```diff
@@ -154,3 +173,140 @@ impl std::fmt::Debug for SendError {
 }
 
 impl std::error::Error for SendError {}
+
+struct ChannelState {
+    shared: Arc<ChannelSharedState>,
+    has_ref: AtomicBool,
+}
+
+impl ChannelState {
+    fn reference<'a, C: Context<'a>>(self: &Self, cx: &mut C) {
```

Comment:
Nitpick: `self: &Self` is idiomatically written `&self`.

Reply:
Ack!